app/helpers/batch_helper.rb

```diff
@@ -69,19 +99,19 @@ def batch_progress_card(batch)
       },
       {
         key: { text: 'Number of CSV rows' },
-        value: { text: batch.rows.count }
+        value: { text: batch.data&.count || 0 }
```

Comment:
I think we should stick to accessing through `#rows` becasue it is lazy there shouldn't be a performance issue